packages/amplify_api/lib/src/graphql/graphql_response_decoder.dart

```diff
@@ -54,9 +54,11 @@ class GraphQLResponseDecoder {
           recoverySuggestion:
               "Current GraphQLResponse is non-nullable, please ensure item exists before fetching");
     }
+    if (request.variables['limit'] != null) {
```

Comment:
callout here: Throwing the limit in the JSON was a lazy way to pass the limit to `PaginatedResult`, which will they use it to make next request. Would it be better to `PaginatedResult` more explicitly somehow? Maybe it could have a setter?